packages/devtools-reps/src/object-inspector/utils/client.js

```diff
@@ -87,11 +89,18 @@ async function getPrototype(
 
 async function getFullText(
   longStringClient: LongStringClient,
-  object: NodeContents
+  item: Node
 ): Promise<{ fullText?: string }> {
-  const { initial, length } = object;
+  const { initial, fullText, length } = getValue(item);
 
   return new Promise((resolve, reject) => {
+    // Return fullText property if it exists so that it can be added to the
+    // loadedProperties map.
+    if (nodeHasFullText(item)) {
```

Comment:
could we do this outside of the Promise callback ? 
```js
if (nodeHasFullText(item)) {
  return Promise.resolve({ fullText })
}

return new Promise(…
```

It feels a bit easier to follow

Reply:
indeed 👍